### PR TITLE
Edit user details

### PR DESCRIPTION
**Pull request by azayneeva**
Relates to issue #122 
Here we can edit user details, update information in the database.

### CODE REVIEW

File: react-backend/client/src/components/admin/adminUserDetailsPage.js [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+          this.props.history.push('/logincb');
+        } else if (error === '500') {
+          this.props.history.push('/internalServerError');
+        }
+      });
+  };
+
+  passwordError = <span>The password is incorrect. Please try again.</span>;
+
+  renderAuthenticated = () => {
```
Comment (finnhodgkin) [inline react-backend/client/src/components/admin/adminUserDetailsPage.js]: I think this component should be split into multiple functional components as you did with the activities page.

File: react-backend/client/src/components/admin/adminUserDetailsPage.js [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    const submitHandler =
+      this.state.userFullName &&
+      this.state.sex &&
+      this.state.yearOfBirth &&
+      this.state.email
+        ? this.handleSubmit
+        : this.handleEmptySubmit;
```
Comment (finnhodgkin) [inline react-backend/client/src/components/admin/adminUserDetailsPage.js]: :+1: :+1:

File: react-backend/database/queries/getUserDetails.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  SELECT *
+  FROM users
+  WHERE cb_id = $1 AND id = $2`;
+
+const getUserDetails = (cbId, userId) =>
+  new Promise((resolve, reject) => {
+    dbConnection
+      .query(getUserDetailsQuery, [cbId, userId])
+      .then(res => {
+        // if (res.rowCount === 0) {
```
Comment (finnhodgkin) [inline react-backend/database/queries/getUserDetails.js]: Remove the commented out code

File: react-backend/database/queries/listAllUsers.js [hunk 4]
```diff
@@ -10,9 +10,9 @@ const getUserList = cbId =>
     dbConnection
       .query(getAllUsersQuery, [cbId])
       .then(res => {
-        if (res.rowCount === 0) {
-          return reject('No user found');
-        }
+        // if (res.rowCount === 0) {
```
Comment (finnhodgkin) [inline react-backend/database/queries/listAllUsers.js]: And again

File: react-backend/database/queries/putNewUserDetails.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+      .query(putNewUserDetailsQuery, [
+        cbId,
+        userId,
+        fullName,
+        sex,
+        yearOfBirth,
+        email
+      ])
+      .then(res => {
+        if (res.rowCount === 0) {
```
Comment (finnhodgkin) [inline react-backend/database/queries/putNewUserDetails.js]: I prefer `if (!res.rowCount) return reject('No user found');`

File: react-backend/app.js [hunk 6]
```diff
@@ -6,6 +6,8 @@ const cookieParser = require('cookie-parser');
 const bodyParser = require('body-parser');
 
 const qrGenerator = require('./routes/qr_generator');
+const userDetails = require('./routes/user_details');
+const newUserDetails = require('./routes/new_user_details');
```
[the remark this answers is not in the document]
Comment (azayneeva) [reply react-backend/app.js]: updated :)

File: react-backend/client/src/App.js [hunk 7]
```diff
@@ -182,6 +183,18 @@ class App extends Component {
               }
             />
 
+            <Route
+              exact
+              path="/admin/user/*"
```
[the remark this answers is not in the document]
Comment (azayneeva) [reply react-backend/client/src/App.js]: done by you :)

File: react-backend/client/src/components/admin/adminUserDetailsPage.js [hunk 8]
```diff
@@ -0,0 +1,270 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { Button } from '../visitors/button';
+import { Logoutbutton } from '../visitors/logoutbutton';
+
+export class AdminUserDetailsPage extends Component {
+  constructor(props) {
+    super(props);
+    const userId = this.props.location.pathname.split('/').reverse()[0];
```
[the remark this answers is not in the document]
Comment (azayneeva) [reply react-backend/client/src/components/admin/adminUserDetailsPage.js]: done :)

File: react-backend/client/src/components/admin/adminUserDetailsPage.js (hunk 1 above)
Comment (azayneeva) [reply react-backend/client/src/components/admin/adminUserDetailsPage.js]: as we agreed we leave it for now :)

File: react-backend/database/queries/getUserDetails.js (hunk 3 above)
Comment (azayneeva) [reply react-backend/database/queries/getUserDetails.js]: done :)

File: react-backend/database/queries/listAllUsers.js (hunk 4 above)
Comment (azayneeva) [reply react-backend/database/queries/listAllUsers.js]: done

File: react-backend/database/queries/putNewUserDetails.js (hunk 5 above)
Comment (azayneeva) [reply react-backend/database/queries/putNewUserDetails.js]: done :)

**Comment by azayneeva** [comment]
I will close this pr as I am pushing James' branch with these fixes and with some new changes

**Pull request closed without merging**